packages/transaction-controller/src/utils/simulation.ts

```diff
@@ -148,13 +144,23 @@ export async function getSimulationData(
   } catch (error) {
     log('Failed to get simulation data', error, request);
 
-    const rawError = error as { code?: number; message?: string };
+    let simulationError = error as SimulationError;
+
+    if (
+      REVERTED_ERRORS.some((revertErrorMessage) =>
+        simulationError.message?.includes(revertErrorMessage),
+      )
```

Comment:
This might be more readable as a regex:

```
const REVERTED_ERROR_REGEX = /execution reverted|insufficient funds for gas/i;
```

Then you can write:
```
if(REVERTED_ERROR_REGEX.test(error.message)) {
  ...
}
```